telemetry_package/main.py

```diff
@@ -1,33 +1,145 @@
+        )
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists.')
+            raise FileExistsError
+            
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Accept the connection
+            self.client_sock, _ = self.server_sock.accept()
+            print('Connection accepted. Creating subscription')
+
+            # After connecting, create two subscriptions: one to the '/rosout' topic and one to the 'control' topic
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.rosout_callback, 
+                qos_profile = qos_profile)
+            
+            self.rosout_subscription
+            print('Created subscription to "/rosout".')
+
+            # Commenting this out for test purposes
+            # self.control_subscription = self.create_subscription(
+            #     Log,
+            #     'control',
+            #     self.control_subscription_callback, 
+            #     qos_profile = qos_profile)
+
+            # self.control_subscription 
+            print('Created subscription to "control".')
+
+
+            self.heartbeat_subscription = self.create_subscription(
+               Log,
+               'heartbeat', 
+                qos_profile = qos_profile)
+            
+            self.heartbeat_subscription
+            print('created subscription to "heartbeat".')
+
+            
+
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(DEFAULT_UNIX_SOCKET_PATH)
+        
+
+    '''
+    @brief This function sanitizes incoming messages so that they do not collide with JSON format guidelines
+
+    @param msg message to sanitize
+
+    @return str
+    '''
+    def sanitize_message(self, msg : str)->str:
+        try: 
+            msg = re.sub("[\{\}:']", "", msg) 
+            return msg 
+        except Exception as e: 
+            print(f'Error occurred during string sanitization: {e}')
+
+
+    '''
+    @brief callback of the subcriptions. 
+
+    This function takes the content of a log message from the /rosout topic, sanitizes it (altough still quite dirtily) and sends it to the ground station webapp via UNIX socket. 
+
+    @param log_msg the message passed by the topic
+    '''
+    def rosout_callback(self, log_msg)->None:
+        try: 
+            log_msg.msg = self.sanitize_message(log_msg.msg)
+
+            #@di-math suggests that this change (#b"\x17") fixes the JSON strings being stitched togeher error 
+            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  log_msg.msg}).encode() + b"\x17")
```

Comment:
I know I have said, that I would leave the ETB out in production. I have actually played around a bit and it would certainly just much cleaner in safer to _let it in_, also in production.
Sorry about the confusion.